code/src/store/slices/snackbarSlice.ts

```diff
@@ -0,0 +1,68 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { sliceNames } from "@/store/constants";
+import { useAppSelector } from "@/hooks/use-redux/useRedux";
+import { TimerId } from "@/types/common";
+import {
+  BaseSnackbarConfig,
+  SnackbarConfigWithTimeout
+} from "@/types/snackbar.types";
+import { AppDispatch } from "@/store/store";
+
+const DEFAULT_AUTOHIDE_DURATION = 3000;
+
+type SnackbarState = {
+  isOpen: boolean;
+  config: BaseSnackbarConfig;
+  _timerId?: TimerId;
+};
+
+const initialState: SnackbarState = {
+  isOpen: false,
+  config: {} as BaseSnackbarConfig
+};
+
+const snackbarSlice = createSlice({
+  name: sliceNames.snackbar,
+  initialState,
+  reducers: {
+    openSnackbar: (state, action: PayloadAction<BaseSnackbarConfig>) => {
+      state.isOpen = true;
+      state.config = action.payload;
+      if (state._timerId) {
+        clearTimeout(state._timerId);
+        delete state._timerId;
+      }
+    },
+    closeSnackbar: (state) => {
+      state.isOpen = false;
+      if (state._timerId) {
+        clearTimeout(state._timerId);
+        delete state._timerId;
+      }
```

Comment:
I suggest to create some utility function which used to encapsulate logic for clearing the timer and reducing code duplication. can create above
It's repeated also for `openSnackbar` as you can see

Reply:
Created utility private function for it

Reply:
> Created utility private function for it

I thought about separate `fn` above. But your case also make sense